### ISSUE TITLE
Unresolved import `std::slice::Join` error with 'discovery compile'

### ISSUE THREAD

**Issue by lauracw100** [enigmampc/discovery-cli#13]
Hey, I updated to the latest discovery-cli version at 0.0.28 and then started getting this error on `discovery compile`:

```
Compiling eng-wasm v0.1.6
error[E0432]: unresolved import 'self::std::slice::Join'
  --> /Users/lauraweindorf/.cargo/registry/src/github.com-1ecc6299db9ec823/eng-wasm-0.1.6/src/internal_std.rs:11:9
   |
11 | pub use self::std::slice::Join;
   |         ^^^^^^^^^^^^^^^^^^^^^^ no 'Join' in 'slice'

error[E0432]: unresolved import 'self::std::slice::Join'
  --> /Users/lauraweindorf/.cargo/registry/src/github.com-1ecc6299db9ec823/eng-wasm-0.1.6/src/internal_std.rs:11:9
   |
11 | pub use self::std::slice::Join;
   |         ^^^^^^^^^^^^^^^^^^^^^^ no 'Join` in `slice'

error: aborting due to previous error

For more information about this error, try 'rustc --explain E0432'.
error: Could not compile 'eng-wasm'.
```

I noticed that this version of `discovery-cli` is using a later version of `eng-wasm` (0.1.6) as opposed to 0.1.3. NOTE: the `simple_addition` secret contract's `Cargo.toml` is still using the earlier versions. Not sure if those versions should be updated to 0.1.6 though.

```
[package]
name = "contract"
version = "0.1.0"

[dependencies]
eng-wasm = "0.1.3"
eng-wasm-derive = "0.1.3"

[lib]
crate-type = ["cdylib"]

[profile.release]
panic = "abort"
lto = true
opt-level = "z"
```

I did some research on `std::slice::Join` and noticed that it's experimental and thought it might be in a more recent nightly build than what's being used in the latest release of `discovery-cli`.

After changing `constants.js` to use the latest rust nightly build and recreating my project with `discovery init`, it's now using the more recent toolchain (rustc 1.39.0 and nightly 09-20-2019 build). The `simple_addition` project compiles. I was able to migrate and run `discovery test` successfully too.

Not suggesting what the correct solution is, but just wanted to let you guys know what my experience was in case that's helpful.

I suspect devs will start having this issue after updating to the 0.0.28 version, unless something's wacky in my environment, which is totally possible :-P.

**Comment by ainsleys** [comment]
cc @lacabra FYI

**Comment by lacabra** [comment]
Thanks @lauraweindorf. This issue has been fixed in #14 (matching the steps you outline above), and version `0.0.29` has been published to npm. In order for this to take effect, you need to install the new version of `discovery-cli` and start fresh with a new project.

Currently, there is no easy way to update an existing `discovery` project to a newer version, that's an enhancement that we're tracking in #17

**Comment by lauracw100** [comment]
Awesome, thanks @lacabra!
